app/policies/transaction_policy.rb

```diff
@@ -4,6 +4,11 @@ def create?
   end
 
   def update?
+    parent_activity = record.parent_activity
+    organisation = parent_activity.organisation
+    fund = parent_activity.fund? ? parent_activity : parent_activity.parent_activities.select { |a| a.level = :fund }
```

Comment:
This is the second place I've seen this logic. Think this might belong in the `Activity` model, but not 100% sure 🤔

Reply:
Ah I see what you mean -- turning this into `Activity#fund?`, rather than changing the machinery of what it's doing with the database :)